analytics.go

```diff
@@ -303,45 +308,116 @@ func (c *client) sendAsync(msgs []message, wg *sync.WaitGroup, ex *executor) {
 	}
 }
 
-// Send batch request.
-func (c *client) send(msgs []message) {
+//Split based on Anonymous ID
+func (c *client) getNodePayload(msgs []message) map[int][]message {
+	nodePayload := make(map[int][]message)
+	totalNodes := c.totalNodes
+	for _, msg := range msgs {
+		anonymousId := gjson.GetBytes(msg.json, "anonymousId").String()
+		hashInt := crc32.ChecksumIEEE([]byte(anonymousId))
+		nodePayload[int(hashInt)%totalNodes] = append(nodePayload[int(hashInt)%totalNodes], msg)
+	}
+	return nodePayload
+}
+
+func (c *client) getRevisedMsgs(nodePayload map[int][]message, startFrom int) []message {
+	msgs := make([]message, 0)
+	for k, v := range nodePayload {
+		if k >= startFrom {
+			for _, msg := range v {
+				msgs = append(msgs, msg)
+			}
+		}
+	}
+	return msgs
+}
+
+func (c *client) setNodeCount() {
 	const attempts = 10
+	for i := 0; i < attempts; i++ {
+		url := c.Endpoint + "/cluster-info"
+		req, err := http.NewRequest("GET", url, bytes.NewReader([]byte{}))
+		if err != nil {
+			c.errorf("creating request - %s", err)
+			continue
+		}
+
+		req.Header.Add("User-Agent", "analytics-go (version: "+Version+")")
+		req.SetBasicAuth(c.key, "")
+
+		res, err := c.http.Do(req)
+
+		if err != nil {
+			c.errorf("sending request - %s", err)
+			continue
+		}
+		if res.StatusCode == 200 {
+			body, err := ioutil.ReadAll(res.Body)
+			if err == nil {
+				c.totalNodes = int(gjson.GetBytes(body, "nodeCount").Int())
```

Comment:
Should we make the default value of totalNodes as 1, as later we are using it in modulus operation for determining the node.

Reply:
Set the default case to 1 in case we get an error